### ISSUE TITLE
ag should include line number after the filename

### ISSUE THREAD

**Issue by ssbarnea** [ggreer/the_silver_searcher#1223]
In order to improve accessibility of the tool and allow users to open files at exact position where the match was found, ag should include the linenumber and eventually even the column number just after the filename, using the `:` separator. 

This notation is something we can already call defacto standard for most, in not all tools like compiliers, linters and other tools returning matches.

Current behaviour:
```
ag "yncing the local repository to the remote host"                                                                                                                                                          [10:07:11]
rtox/rtox.py
165-
166-    # Clone the repository we're working on to the remote machine.
167:    print('Syncing the local repository to the remote host...')
168-    subprocess.check_call([
169-        'rsync',
```

Proposed behaviour:
```
ag "Syncing"                                                                                                                                                          [10:07:11]
rtox/rtox.py:167:11
165-
166-    # Clone the repository we're working on to the remote machine.
167:    print('Syncing the local repository to the remote host...')
168-    subprocess.check_call([
169-        'rsync',
```

Note: on multiple matches on same file, filename should be repeated.

**Comment by alphapapa** [comment]
ag has lots of output options, including for the filename.  Have you looked at the help and man page?

Unfortunately, it still seems to have bugs that cause it to not output the filename even when `--filename` is specified...

**Comment by ssbarnea** [comment]
I think that adding the line number after the filename would greatly increase the usability of ag as we could click on results and open them in the editor directly.

<img width="416" alt="ag-search-results" src="https://user-images.githubusercontent.com/102495/41991089-3c16d942-7a3c-11e8-9f71-bf2705f37228.png">

**Comment by alphapapa** [comment]
Have you considered the `--ackmate` option?  Or you might consider using another tool, like `rg`.

**Comment by wind2412** [comment]
I agree, due to the fact that I often open a file using Iterm2 on macOS, I need to scroll to the line number on my own >_<. In fact `grep`'s output is better for opening a file but it's slower than `ag`.

**Comment by digitalmaster** [comment]
Just following up on this.. is there really still no way to do this yet? Would love to be able to click on a file and be taken right to the result. 😆

**Comment by ssbarnea** [comment]
Sorry but I switched to [ripgrep](https://github.com/BurntSushi/ripgrep) one or two years ago, nothing last forever.

**Comment by digitalmaster** [comment]
Indeed. Thanks.. will give that a shot 👍

**Comment by 64kramsystem** [comment]
It a bit disappointing that this feature has been ignored.

For those coming here for a solution, a tiny little bit of scripting will do - the output structure is regular, so it can be trivially processed. This is a possible approach:

```sh
ag "$pattern" | perl -lne 'if (/^(.+?):(\d+)/ && !$files{$1}) { print "$1:$2"; $files{$1} = "printed" }'
```

The Perl script extracts filename and line, and, if they aren't in a hashmap that tracks them, prints the filename and line, and puts the filename into the hashmap (so that it isn't printed again).

**Comment by zweix123** [comment]
> Sorry but I switched to [ripgrep](https://github.com/BurntSushi/ripgrep) one or two years ago, nothing last forever.

How did `rg` achieve this goal?
